src/compiler/scala/tools/nsc/tasty/TreeUnpickler.scala

```diff
@@ -0,0 +1,1126 @@
+package scala.tools.nsc.tasty
+
+import TastyRefs._
+
+import scala.annotation.switch
+import scala.collection.mutable
+import scala.reflect.io.AbstractFile
+import scala.reflect.internal.Variance
+import scala.util.chaining._
+
+/** Unpickler for typed trees
+ *  @param reader              the reader from which to unpickle
+ *  @param splices
+ */
+class TreeUnpickler[Tasty <: TastyUniverse](
+    reader: TastyReader,
+    nameAtRef: NameRef => TastyName,
+    splices: Seq[Any])(implicit
+    val tasty: Tasty) { self =>
+  import tasty._, FlagSets._
```

Comment:
main aim here is that `tasty` provides some type aliases and we don't directly select from `tasty.symbolTable`, not  yet converted to abstract types due to directly selecting on symbols here